##### 1. Find out how many males and females participated in the test.
##### 2. What do you think about the students' parental level of education?
##### 3. Who scores the most on average for math, reading and writing based on
● Gender
● Test preparation course
##### 4. What do you think about the scoring variation for math, reading and writing based on
● Gender
● Test preparation course
##### 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this. 

In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [5]:
stud_data = pd.read_csv('StudentsPerformance.csv')

In [6]:
stud_data

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

In [7]:
stud_data.isna().values.any()

False

In [8]:
stud_data.dtypes

gender                         object
race/ethnicity                 object
parental level of education    object
lunch                          object
test preparation course        object
math score                      int64
reading score                   int64
writing score                   int64
dtype: object

### 1. Number of male and female students  

In [9]:
stud_data[stud_data['gender']=='female'].shape[0]

518

In [10]:
stud_data[stud_data['gender']=='male'].shape[0]

482

518 female and 482 male students participated in the test


### 2. Parental level of  Education 

In [11]:
stud_data['parental level of education'].mode()

0    some college
dtype: object

Most of the students' parental level of education is some college

### 3. Average scores with respect to gender and test preparation course 

In [12]:
stud_gr_gender = stud_data.groupby('gender').mean()
stud_gr_gender

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

Female students have higher average score in reading and writing than the male students. But male students have higher average score in Maths.

In [13]:
stud_gr_tpc = stud_data.groupby('test preparation course').mean()
stud_gr_tpc

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

Students who completed the test preparation course have higher average score in Math, reading and writing.

### 4. Variation of  scores with respect to gender and test preparation course 

In [14]:
stud_gr_gender_var = stud_data.groupby('gender').std()
stud_gr_gender_var

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

Variation of scores for female students is higher than male students in all the three categories.

In [15]:
stud_gr_tpc_var = stud_data.groupby('test preparation course').std()
stud_gr_tpc_var

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

Among students who have completed the test preparation course, the variation of scores is less in all the three subjects.

### 5. Top 25% Students in Maths

In [16]:
np.quantile(stud_data['math score'], 0.75)

77.0

Students who scored more than 77 in Maths constitute to the top 25%.

In [17]:
stud_data[stud_data['math score'] >= 77.0].sort_values('math score', ascending=False)

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
998    male        group E          associate's degree  free/reduced   
997    male        group D                some college      standard   
996    male        group A                some college      standard   
517  female        group E          associate's degree      standard   
..      ...            ...                         ...           ...   
854    male        group D            some high school      standard   
855    male        group E          associate's degree  free/reduced   
856    male        group E            some high school      standard   
415  female        group B             master's degree  free/reduced   
414  female        group A                 high school  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
998               completed         100            100             93  
997               completed         100             97             99  
996               completed         100             96             86  
517                    none         100            100            100  
..                      ...         ...            ...            ...  
854               completed          77             68             69  
855               completed          77             69             68  
856               completed          77             76             77  
415               completed          77             97             94  
414               completed          77             88             85  

[255 rows x 8 columns]